core/arch/aarchxx/mangle.c

```diff
@@ -1143,10 +1143,14 @@ reinstate_it_blocks(dcontext_t *dcontext, instrlist_t *ilist, instr_t *start,
             /* A label instruction may be used as a cti target, so we stop
              * the IT block on label instructions.
              */
-            !instr_is_label(instr) &&
-            /* Do not put OP_b exit cti into block: patch_branch can't handle */
-            instr_get_opcode(instr) != OP_b &&
-            instr_get_opcode(instr) != OP_b_short;
+            !instr_is_label(instr);
+        if (for_mangle) {
```

Comment:
Yeah I would just put this back to how it was and keep the (predicated) direct branches outside of the IT blocks.